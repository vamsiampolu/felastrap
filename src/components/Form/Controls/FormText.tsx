import { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";

const textRule = () => ({
  marginTop: "0.25rem",
  fontSize: "0.875em",
  color: "rgba(33, 37, 41, 0.75)",
});

export type Children = ComponentPropsWithoutRef<"div">["children"];

export const FormText = ({
  children,
  className = "",
}: {
  className?: string;
  children: Children;
}) => {
  const { css } = useFela();
  return (
    <div
      data-testid="form-text"
      className={`${css(textRule)} form-text ${className}`.trim()}
    >
      {children}
    </div>
  );
};
