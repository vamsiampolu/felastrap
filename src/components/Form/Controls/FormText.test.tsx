import { render, screen } from "@testing-library/react";
import { FormText } from "./FormText";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../../test-helpers";
import { IRenderer } from "fela";

describe("components/Form/Controls/FormText", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a form text", async () => {
    const component = <FormText>Form Text</FormText>;

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("form-text")).toBeInTheDocument();

    expect(screen.getByTestId("form-text")).toHaveClass("form-text");
    expect(screen.getByTestId("form-text")).toHaveTextContent("Form Text");

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
