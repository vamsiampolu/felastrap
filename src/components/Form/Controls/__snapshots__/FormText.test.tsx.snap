// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form/Controls/FormText should render a form text 1`] = `
".a {
  margin-top: 0.25rem;
}
.b {
  font-size: 0.875em;
}
.c {
  color: rgba(33, 37, 41, 0.75);
}


<div data-testid="form-text" class="a b c form-text">Form Text</div>
;
"
`;
